lib/blockscore/base.rb

```diff
@@ -15,11 +17,11 @@ def initialize(options = {}, &block)
     def attributes
       return @attributes if @loaded
       force!
-      @attributes
+      attributes
     end
 
     def id
-      @attributes.fetch(:id, nil)
+      attributes.fetch(:id, nil)
```

Comment:
> This could actually be a mutation:
> 
> ```ruby
> foo.fetch(bar, nil) -> foo[bar]
> ```

Yes, exactly. This is a common scenario. While I prefer `Hash#fetch` for most things, I also believe that if your use case *exactly* corresponds to the API of a given method, you should use that method rather than recreating it's semantics using something else.